python/cudf/cudf/core/column/datetime.py

```diff
@@ -242,14 +242,19 @@ def normalize_binop_value(self, other: DatetimeLikeScalar) -> ScalarLike:
         if isinstance(other, (cudf.Scalar, ColumnBase, cudf.DateOffset)):
             return other
 
-        if isinstance(other, datetime.datetime):
-            other = np.datetime64(other)
-        elif isinstance(other, datetime.timedelta):
-            other = np.timedelta64(other)
-        elif isinstance(other, pd.Timestamp):
+        if isinstance(other, pd.Timestamp):
+            if other.tz is not None:
+                raise NotImplementedError(
+                    "Cannot perform binary operation on timezone-naive columns"
+                    " and timezone-aware timestamps. "
+                )
             other = other.to_datetime64()
         elif isinstance(other, pd.Timedelta):
             other = other.to_timedelta64()
+        elif isinstance(other, datetime.datetime):
```

Comment:
What about TZ-aware `datetime.datetime` objects? e.g.
```python
>>> import datetime
>>> datetime.datetime.now(datetime.timezone.utc)
datetime.datetime(2023, 8, 7, 20, 47, 48, 483493, tzinfo=datetime.timezone.utc)
```

Do those need to error, too? On conversion to `np.datetime64`, I get a warning. Is that sufficient?

```python
>>> np.datetime64(datetime.datetime(2023, 8, 7, 20, 47, 48, 483493, tzinfo=datetime.timezone.utc))
<stdin>:1: DeprecationWarning: parsing timezone aware datetimes is deprecated; this will raise an error in the future
numpy.datetime64('2023-08-07T20:47:48.483493')
```

Reply:
I think we should raise for it too. Good catch, I thought `datetime` was tz agnostic.

Reply:
@bdice I updated the code to handle `datetime.datetime` objects aswell.